tests/test_pagination.py

```diff
@@ -217,15 +216,18 @@ def sqla_14_all_users_with_notes_count(paginate: Paginate = Depends()):
         )
         return paginate(query, scalars=False)
 
-    def query_count(session: Session, query: Select) -> int:
-        return session.execute(select(func.count()).select_from(user_cls)).scalar()
+    def count_user_notes(user_id: int, session: Session = Depends()) -> int:
+        return session.execute(
+            select(func.count(note_cls.id)).where(note_cls.user_id == user_id)
+        ).scalar()
```

Comment:
@vicrep I strongly believe the query counter function should be a FastAPI dependency to support this ^ 

What do you think?

Reply:
@hadrien I agree in the context of SQLA2 language. The nice thing though is that making it a dependency still allows us to do pure counter functions if DI is not needed (basically, if you don't need it to be a dependency, then pretend it isn't one :) )